docs/references/architecture.mdx

```diff
@@ -0,0 +1,103 @@
+
+
+## Overview
+
+Restate is implemented with a three-layered architecture: a control plane, distributed log, and processors.
+
+<img width={"750px"} alt={"Restate Node internal architecture"} src={"/img/architecture/restate_layered_arch.png"}/>
+
+
+### Control plane
+This component stores all metadata about deployments (what services exist behind which endpoint addresses / URLs / ARNs) and is responsible for assigning the leaders to log partitions and processors.
+
+
+### Distributed Log a.k.a Bifrost
+Restate uses a distributed log, called Bifrost, to durably record all events in the system before acting on them, similar to the function of a write-ahead log (WAL) in a database system.
+The design of Bifrost is based on the idea of a virtual consensus, as described in the [Delos paper](https://www.usenix.org/system/files/osdi20-balakrishnan.pdf).
+The consensus algorithm powering Bifrost is based on [Flexible Paxos](https://fpaxos.github.io/).
+
+To support scale-out operations, Restate splits the services key-space across multiple partitions, backed by logs. Currently, we map one partition to one log, though this relationship may change in the future. Thus you will see references to both partition ids and log ids, depending on context, so be aware that these are distinct concepts even if they might be the same value.
+
+
+### Processors
+
+The Processors (also called Partition Processors) receive invocations, process events from the durable log, and interact with the services/functions containing the application/workflow logic. The Processors encapsulate the state machines for durable execution and invocation life-cycles.
+
+The Processors maintain the “state of the world” which is derived from the events in the log. That state describes what invocations are ongoing, what is in the journal of each invocation, futures/promises, and the actual state of virtual objects. This state can be thought of as a materialized view that is deterministically derived from the log. The state is stored in RocksDB and periodically snapshotted (to persistent volume or object store), to bound the number of events to re-apply during a failure-recovery-catchup cycle.
+
+## Nodes and roles
+
+You'll see many mentions of the terms server and node throughout this documentation. Generally, we use the term "server" to refer to a running instance of the `restate-server` binary, although in some instances this might also refer to the entire Restate cluster which acts as a single entity. This binary can host multiple functions. When you start a single-node Restate server, for example when doing some local development or testing, you are hosting all the essential features in a single process. These include accepting incoming requests, durably recording events, processing work (delegating invocations to services, handling key-value operations), as well as maintaining metadata used internally by the system.
```

Comment:
Instead of "single-node Restate server" maybe just "Restate server"? Otherwise, people might wonder whether there exists a multi-node Restate server.